addons/web/static/src/core/overlay/overlay_container.js

```diff
@@ -9,14 +9,21 @@ export class OverlayContainer extends Component {
 
     setup() {
         this.root = useRef("root");
+        this.state = useState({ rootEl: null });
+        useEffect(
+            () => {
+                this.state.rootEl = this.root.el;
+            },
+            () => [this.root.el]
+        );
```

Comment:
This implementation is simpler, but this necessarily triggers 2 renderings whenever the component is mounted.
The current implementation of the PR guarantees only 1 render when there's no overlay, but 2 when mount happens with some registered overlays.

Are you ok with 2 renderings all the time at mount? Visually this is not prone to flickers, in theory this could be less efficient for OWL, but probably not a problem enough to care compared to code maintenance with simpler code you suggest?